driver-core/src/main/com/mongodb/internal/operation/ClientBulkWriteOperation.java

```diff
@@ -258,6 +329,80 @@ private Integer executeBatch(
         }
     }
 
+    /**
+     *  @param finalCallback A callback that accepts the start model index of the next batch, provided that the operation
+     * {@linkplain ExhaustiveClientBulkWriteCommandOkResponse#operationMayContinue(ConcreteClientBulkWriteOptions) may continue}
+     * and there are unexecuted {@linkplain ClientNamespacedWriteModel models} left.
+     */
+    @Nullable
+    private void executeBatchAsync(
+            final int batchStartModelIndex,
+            final WriteConcern effectiveWriteConcern,
+            final AsyncWriteBinding binding,
+            final ResultAccumulator resultAccumulator,
+            final SingleResultCallback<Integer> finalCallback) {
+        List<? extends ClientNamespacedWriteModel> unexecutedModels = models.subList(batchStartModelIndex, models.size());
+        assertFalse(unexecutedModels.isEmpty());
+        OperationContext operationContext = binding.getOperationContext();
+        SessionContext sessionContext = operationContext.getSessionContext();
+        TimeoutContext timeoutContext = operationContext.getTimeoutContext();
+        RetryState retryState = initialRetryState(retryWritesSetting, timeoutContext);
+        BatchEncoder batchEncoder = new BatchEncoder();
+
```

Comment:
As discussed on the call, instead of removing the line breaks from the async implementation, I’ve added the same line breaks to the sync version for consistency.